AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/OutgoingMutationQueue/OutgoingMutationQueue.swift

```diff
@@ -188,6 +192,12 @@ final class OutgoingMutationQueue: OutgoingMutationQueueBehavior {
                 self.processSyncMutationToCloudResult(result, mutationEvent: mutationEvent, api: api)
         }
         operationQueue.addOperation(syncMutationToCloudOperation)
+
+        let outboxStatusEvent = OutboxStatusEvent(isEmpty: operationQueue.operationCount == 0 ? true : false)
+        let outboxStatusEventPayload = HubPayload(eventName: HubPayload.EventName.DataStore.outboxStatus,
+                                               data: outboxStatusEvent)
+        Amplify.Hub.dispatch(to: .dataStore, payload: outboxStatusEventPayload)
```

Comment:
- can encapsulate into `dispatchOutboxStatusEvent(isEmpty: false)` and called right before adding the operation to operationQueue.
- it should always be true because you are about to add an operation into the operation queue. even if you dispatch after adding it in, most likely the count will be none zero but then you are concurrently executing the operation and performing a check on the operationQueue count.

Reply:
OK. But is this like hardcoded Although we know we are about to add an operation into queue? 😂

Reply:
because the operation queue is not suspended so if you add an operation to the queue, you can immediately assume it is in the process of being executed. like the other scenario, first dispatch the hub event then start the operation (by adding it to the operation queue or unsuspend the operation queue).